bin/analyzeShearTest.py

```diff
@@ -0,0 +1,265 @@
+    filterKey = schema.find("filter").getKey()
+    seeingKey = schema.find("seeing").getKey()
+    shearKey = schema.find("shear").getKey()
+    nsubKey = schema.find("nsub").getKey()
+    nsourceKey = schema.find("nsource").getKey()
+    gKey = schema.find("g").getKey()
+    eAvgKey = schema.find("eAvg").getKey()
+    eStdKey = schema.find("eStd").getKey()
+    e1DevKey = schema.find("e1Dev").getKey()
+    e1StdKey = schema.find("e1Std").getKey()
+    e2DevKey = schema.find("e2Dev").getKey()
+    e2StdKey = schema.find("e2Std").getKey()
+
+    #   open the run_params file for this run and extract what we need.
+    fin = open(os.path.join(args.base_dir, "run_params"), 'r')
+    params = dict()
+    for line in fin:
+        if line.isspace():
+            continue
+        inp = line.split()
+        if inp[0][0] == '#':
+            continue
+        params[inp[0]] = inp[1]
+
+    shear_value = float(params["shear_value"])
+    psf_file = os.path.basename(params["psf_dir"])
+    filter =  int(params["filter"])
+    seeing  = float(params["seeing"])
+    n_subfields = int(params["n_subfields"])
+    n_epochs = int(params["n_epochs"])
+    galaxy_stamp_size = int(params["galaxy_stamp_size"])
+    output_dir = params["output_dir"]
+
+    #   This tells us the algorithm fields we need to extract the ellipticity and error
+    #   The shape type can be "moments" or "ellipticity"
+    #   The error can be extracted from an ellipticity error field or based on the
+    #   signal-to-noise of some other field such as flux and flux error.  In that
+    #   case, the weighting value used is shape_field/(sigma_field/sigma_signal_field)
+    if "shape_field" in params.keys():
+        shape_field = params["shape_field"] + "_"
+        shape_type = params["shape_type"]
+        if "sigma_field" in params.keys():
+            sigma_field = params["sigma_field"]
+            if "sigma_signal_field" in params.keys():
+                sigma_signal_field = params["sigma_signal_field"]
+            else:
+                sigma_signal_field = None
+        else:
+            sigma_field = None
+            sigma_signal_field = None
+    else:
+        #   If shape_field is not in params, the ellipticity and sigma field are here:
+        shape_field = ""
+        shape_type = "ellipticity"
+        sigma_field = "e_sigma"
+        sigma_signal_field = None
+
+    #   Use the butler to get the src.fits file for each subfield/epoch
+    butler = dafPersist.Butler(root=os.path.join(args.base_dir, output_dir))
+
+    count = 0
+    eSum = 0.0
+    e1DevSum = 0.0
+    e2DevSum = 0.0
+    eSumSq = 0.0
+    e1DevSumSq = 0.0
+    e2DevSumSq = 0.0
+    thetaSum = 0.0
+    for subfield in range(n_subfields):
+        for epoch in range(n_epochs):
+            catCount = 0
+            catE1DevSum = 0.0
+            catE2DevSum = 0.0
+            catWeightSum = 0.0
+            sourceCat = butler.get("src", {'subfield':subfield, 'epoch':epoch})
```

Comment:
I always recommend using `immediate=True` in `butler.get` calls; not doing that can delay exceptions until later, where they're harder to figure out (because without it, the butler returns a lazy-loading proxy object).